### DISCUSSION TITLE
Fail2ban works to hard!

### DISCUSSION THREAD

**Discussion by lazat77** [fail2ban/fail2ban#3128] [Q&A] [answered]
I have a problem that f2b reads smtp log files for smtpauth logins. If wrong password is enter the users ip is blocked. This is good for brute forceing but is gives me another problem.  If an user changes their password in hosting controlpanel and skips to update it in email client the users is blocked and they call support. Would it be possible to have f2b to only block if an ip only tests diffent passwords and not block if the samt user, password, ip is the same?

**Comment by sebres** [comment, marked as answer]
> Would it be possible to have f2b to only block if an ip only tests diffent passwords and not block if the samt user, password, ip is the same?

I'm afraid the answer to this is 'no'.
I don't know any service logging wrong passwords, so fail2ban would not be able physically to consider the password is the same.

Conditionally an increase of `maxretry` (e. g. with `findtime`) may help here, but it is not really an option (a botnet can find it out relative simply and make a bruteforce faster, but a legitimate user can anyway cause a ban if the mail client is fast enough to make `maxretry` attempts until the user manage to change the password).

I see only few following options after the password change:
1. the smtp-/mail-service logging different message for the user if it still uses old password (or always the same password).
2. you'd exclude the user for a short time (e. g. up to few hours) in fail2bans jail:
   - either with `ignorecommand` which could check whether the user (e. g. captured with tag `<F-USER>`) was changing the password and ignore it for some short time.
   - or by change of `failregex` in fail2ban (e. g. insert some lookahead RE for ignored user logins)

Basically only option 1 is good but it is not always possible (e. g. expecting enhancement of the service).

**Reply by lazat77** [reply]
Ok, thanks, I guess I have to recompile the smtp module. This should not be to hard as I already have the source build tree for other patches I have installed.
